Expensify/expensify-common
# Acknowledge CRLF as possible EOL sequence for codefence

The back-end uses CRLF (`\r\n`) as EOL sequence for task description (reference [here](https://github.com/Expensify/App/blob/9302ade9988b1cba3e07b48f474bc35a6deea7fb/src/pages/tasks/TaskDescriptionPage.tsx#L50)). In our parser, we only acknowledge LF (`\n`) as a possible EOL after 3 backticks fence, if it's not there, we'll append new line, this leads to a new line being added each time we edit task description.

### Fixed Issues
$ https://github.com/Expensify/App/issues/38001

# Tests
**1. What unit/integration tests cover your change? What autoQA tests cover your change?**

The `ExpensiMark-HTML-test.js` is updated to include the new test cases.

**2. What tests did you perform that validates your changed worked?**

- Run the unit test.
- Test in Expensify App follow these steps:

1. Edit a task description with code fence without line break

````
```task description```
````

2. Save the description
3. Open task descrition edit page again
4. Repeat step 2 and 3 several times
5. Verify that no extra line break is added each time saving the description

# QA

**1. What does QA need to do to validate your changes?**

Test in Expensify App follow these steps:

1. Edit a task description with code fence without line break

````
```task description```
````

2. Save the description
3. Open task descrition edit page again
4. Repeat step 2 and 3 several times
5. Verify that no new description is saved


**2. What areas to they need to test for regressions?**

NA

# Screenshots


https://github.com/Expensify/expensify-common/assets/113963320/8733ce95-7225-4918-88a7-f400dfe97201
